java-samples/corda5-negotiation-cordapp/workflows/src/main/java/com/r3/developers/samples/negotiation/workflows/accept/AcceptFlowRequest.java

```diff
@@ -0,0 +1,114 @@
+        UUID proposalID = request.getProposalID();
+
+        // Getting the old Proposal State as an input state
+        List<StateAndRef<Proposal>> proposalStatAndRef = utxoLedgerService.findUnconsumedStatesByType(Proposal.class);
+        List< StateAndRef<Proposal>> proposalStatAndRefWithId = proposalStatAndRef.stream().
+                filter(it->it.getState().getContractState().getProposalID().equals(proposalID)).collect(Collectors.toList());
+
+        if(proposalStatAndRefWithId.size()!=1) throw  new CordaRuntimeException("Multiple or zero Proposal states with id " + proposalID + "found");
+        StateAndRef<Proposal> proposalStateAndRef = proposalStatAndRefWithId.get(0);
+        Proposal proposalInput = proposalStateAndRef.getState().getContractState();
+
+        // Creating a Trade as an output state
+        Trade output = new Trade(proposalInput.getAmount(),
+                new Member(proposalInput.getBuyer().getName(),proposalInput.getBuyer().getLedgerKey()),
+                new Member(proposalInput.getSeller().getName(), proposalInput.getSeller().getLedgerKey()),
+                proposalInput.getParticipants()
+                );
+
+        Member counterParty= (memberLookup.myInfo().getName().equals(proposalInput.getProposer().getName()))?proposalInput.getProposee(): proposalInput.getProposer();
+
+        //Initiate the transactionBuilder with command to "Accept"
+        UtxoTransactionBuilder transactionBuilder = utxoLedgerService.createTransactionBuilder()
+                .setNotary(proposalStateAndRef.getState().getNotaryName())
+                .setTimeWindowBetween(Instant.now(), Instant.now().plusMillis(Duration.ofMinutes(5).toMillis()))
+                .addInputState(proposalStateAndRef.getRef())
+                .addOutputState(output)
+                .addCommand(new ProposalAndTradeContract.Accept())
+                .addSignatories(output.getParticipants());
+
+
+        // Call FinalizeIOUSubFlow which will finalise the transaction.
+        // If successful the flow will return a String of the created transaction id,
+        // if not successful it will return an error message.
+        try{
+            UtxoSignedTransaction signedTransaction = transactionBuilder.toSignedTransaction();
+            FlowSession counterPartySession = flowMessaging.initiateFlow(counterParty.getName());
+            UtxoSignedTransaction finalizedTransaction= utxoLedgerService.finalize(signedTransaction,List.of(counterPartySession)).getTransaction();
+
+            return  finalizedTransaction.getId().toString();
+        }
+        catch (Exception e){
+            return String.format("Flow failed, message: %s", e.getMessage());
+        }
+    }
+}
+/*
+
+Alice hash: 5B9459F205F0
+
+{
+  "clientRequestId": "flow name",
+  "flowClassName": "com.r3.developers.samples.negotiation.workflows.accept.AcceptFlowRequest",
+  "requestBody": {
+      "proposalID": "9d9d36df-6091-4bcf-8334-5926a6ba422b"
+  }
+}
+ */
```

Comment:
is this comment meant to be here?

member vNode hashes are not necessarily fixed